http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProvider.scala

```diff
@@ -127,6 +127,55 @@ class SamAuthProvider[F[_]: OpenTelemetryMetrics](
     } yield (callerActions, projectCallerActions)
   }
 
+  /**
+   * Get all Sam resource IDs of the given type on which the user is granted permissions. Returns explicitly granted
+   * policies only; users with workspace owner access, for example, can see all runtimes in the workspace, but
+   * getAuthorizedIds[SamRuntimeResourceId] is not guaranteed to return all runtimes in that workspace. Works for any
+   * Sam resource type except SharedApp and similar alternative app types; these require a specific Sam resource object
+   * to determine permissions, as these are stored on the individual records.
+   * @param isOwner whether to return only resources that the user is granted Owner permission on (or the equivalent for
```

Comment:
@Qi77Qi or @jdcanas , do y'all know how the SamResource::policyNames and SamResource::ownerRoleName are used? Is either of them deprecated, or do we use both? Is `policyNames` a complete list of possible policies on the resource, or is it just a list of policies Leonardo cares about? For `WsmResource`, why is the `ownerRoleName` "Owner" not present in `policyNames`?

tl;dr, is my logic here correct?

Reply:
From what I can tell, both fields are used in some cases. WsmResource appears to be configured to hide the Owner role from Leo. This might be an error.

Reply:
I answered these questions to my satisfaction.